openapi/src/parts/controlled_azure_vm.yaml

```diff
@@ -133,6 +130,106 @@ components:
           description: A valid WSM identifier for an Azure Network that corresponds to a valid azure resource
           type: string
           format: uuid
+        vmImage:
+          description: A valid data which identifies the VM image
+          $ref: '#/components/schemas/AzureVmImage'
+        vmUser:
+          description: A valid VM user data
+          $ref: '#/components/schemas/AzureVmUser'
+        customScriptExtension:
+          description: A custom script extension
+          $ref: '#/components/schemas/AzureVmCustomScriptExtension'
+
+    AzureVmImage:
+      description: Data which identifies the VM image. uri parameter is mutual exclusize with the publisher, offer and sku 
+        uri should be used in case we need to use custom image, otherwise publisher, offer, sku should be used to use existing image from the marketplace. |
+        uri for custom image should be in specific format and the rest of parameters will be ignored. Must be in the same region specified. For instance /subscriptions/3efc5bdf-be0e-44e7-b1d7-c08931e3c16c/resourceGroups/mrg-qi-1-preview-20210517084351/providers/Microsoft.Compute/galleries/msdsvm/images/customized_ms_dsvm/versions/0.1.0
+        In case of image from marketplace possible parameters are publisher=microsoft-dsvm, offer=ubuntu-1804, sku=1804-gen2
```

Comment:
if caller defines both, which one will take precedence? can we document that

Reply:
Fixed.